### PR TITLE
IO-700: provide method overload for moveFile

### PR DESCRIPTION
**Pull request by tza**
- added test cases
- updated javadoc for "copy and delete" with setLastModified

### CODE REVIEW

**Comment by tza** [comment]
So far the test cases seem ok for me: 
The failing tests FileUtilsCleanDirectoryTestCase were introduced with commit 0cee29aa4c1818963ed1a55058219282e89d7488 and failed in my workspace before changing anything.
The AbstractInputStreamTest.tearDown IllegalAccess error also happened before, eg. in pull 181. Seems unrelated with my changes.

**Comment by garydgregory** [comment]
> So far the test cases seem ok for me:
> The failing tests FileUtilsCleanDirectoryTestCase were introduced with commit [0cee29a](https://github.com/apache/commons-io/commit/0cee29aa4c1818963ed1a55058219282e89d7488) and failed in my workspace before changing anything.
> The AbstractInputStreamTest.tearDown IllegalAccess error also happened before, eg. in pull 181. Seems unrelated with my changes.

I must have messed up, the build passed for me locally, but, obviously it's broken. I will look...

**Comment by tza** [comment]
Thanks, just let me know if I should rebase and push again. 

ps. I'm pretty new in the open-source / github community, so sorry for any rookie mistake :)

**Review by garydgregory — CHANGES_REQUESTED** [review]
> Thanks, just let me know if I should rebase and push again.
> 
> ps. I'm pretty new in the open-source / github community, so sorry for any rookie mistake :)

Yes, please rebase on master, and we should see a green build (aside from expected EA failures)

File: src/main/java/org/apache/commons/io/FileUtils.java [hunk 1]
```diff
@@ -2172,12 +2177,39 @@ public static void moveDirectoryToDirectory(final File src, final File destDir,
...
+     *
+     * @param srcFile the file to be moved.
+     * @param destFile the destination file.
+     * @param preserveFileDate true if the file date of the "copy and delete" should be the same as the original when
+     *            destination is on another file system. Param is not used if destination is on same file system.
+     * @throws NullPointerException if any of the given {@code File}s are {@code null}.
+     * @throws FileExistsException if the destination file exists.
+     * @throws IOException if source or destination is invalid.
+     * @throws IOException if an error occurs or setting the last-modified time didn't succeeded.
+     * @since 1.4
```
[the remark this answers is not in the document]
Comment (tza) [reply src/main/java/org/apache/commons/io/FileUtils.java]: Rebased and fixed since tag.

**Pull request merged**